tools/readtip/main.go

```diff
@@ -40,16 +44,32 @@ func init() {
 	flag.Parse()
 }
 
-func main() {
+func readStateDBPath() string {
+	if _stateDBPath != "" {
+		return _stateDBPath
+	}
 	cfg, err := config.New([]string{_overwritePath, _secretPath}, []string{})
```

Comment:
nit: can put these logics (check `_stateDBPath`, if empty then New() from `_overwritePath, _secretPath`) into a utility func
```
cfg, err := generateConfig()
```
then directly use it in main()
```
store, err := db.CreateKVStore(cfg.DB, cfg.Chain.TrieDBPath)
```